rrtmgp/mo_gas_optics_rrtmgp.F90

```diff
@@ -1513,10 +1537,12 @@ function compute_optimal_angles(this, optical_props, optimal_angles) result(err_
     integer  :: ncol, nlay, ngpt
     integer  :: icol, ilay, igpt, bnd
     real(wp) :: t, trans_total
+    integer, allocatable  :: bands(:)
```

Comment:
@vectorflux Why not use `integer :: bands(optical_props%get_ngpt))` ? Saves explicit de/allocation below.

Reply:
`integer :: bands(optical_props%get_ngpt))` fails with the CCE 14.0.3 compiler, so it needs to be allocated explicitly, at least for the time being.

Reply:
@vectorflux it might make sense to document the workaround with
```fortran
#if defined _CRAYFTN && _REALEASE_MAJOR == 14 && _RELEASE_MINOR == 0 && _RELEASE_PATCHLEVEL == 3
# define CRAY_WORKAROUND
#endif

#ifdef CRAY_WORKAROUND
integer, allocatable  :: bands(:)
#else
integer :: bands(optical_props%get_ngpt))
#endif
```